### PR TITLE
Vulkan: Use update_deps.py to get correct version of headers.

### LINKED ISSUE #1179: Vulkan header fails because of '-Wno-stringop-truncation'
[logs.zip](https://0x0.st/zb4_.zip)

### PR DESCRIPTION
**Pull request by 1480c1**
Closes #1179 (?)
@Selur and @Helenerineium  , can you confirm if this fixes the error?

I think there might be a problem with jq still because of #1179 and #1172 . Will need to investigate

### CODE REVIEW

**Comment by wiiaboo** [comment]
Alternatively, you could do like mpv and run the python tool inside vulkan-loader and then compile/install vulkan-headers in it: https://github.com/mpv-player/mpv/commit/4e9dc60d9e7728cddfc901a8b0b7048eeebed5e6

**Comment by wiiaboo** [comment]
I mistook shaderc with vulkan-loader, but the same tool is there as well: https://github.com/KhronosGroup/Vulkan-Loader/blob/master/scripts/update_deps.py

**Comment by 1480c1** [comment]
I might actually do that since it seems that that would be more reasonable and maintainable

**Comment by 1480c1** [comment]
Problem with using the python script, it uses cmake's default generator, which for some, happens to be Visual Studio.
I've tried to change the settings using the json, but I can't seem to get the CMAKE_INSTALL_PREFIX to be set to `LOCALDESTDIR` but it doesn't parse it like normal.

Guess I have to learn python.

**Comment by wiiaboo** [comment]
https://github.com/KhronosGroup/Vulkan-Loader/blob/master/scripts/update_deps.py#L565-L571

Just use `--no-build` and move vulkan-header compilation/installatio to inside vulkan-loaders.
There's also msvc-specific options for the update_deps.py build part, so it's better to just not build with it.

**Comment by 1480c1** [comment]
Yea, that's what I'm thinking to do

File: build/media-suite_compile.sh [hunk 1]
```diff
@@ -1846,39 +1846,32 @@ if [[ $mpv != "n" ]] && pc_exists libavcodec libavformat libswscale libavfilter;
...
-        _check=(vulkan/vulkan.h)
-        if do_vcs "https://github.com/KhronosGroup/Vulkan-Headers.git#tag=${_Vulkan_Version:-v1.1.106}" vulkan-headers; then
+    _check=(libvulkan.a vulkan.pc)
+    if ! mpv_disabled vulkan &&
+        do_vcs "https://github.com/KhronosGroup/Vulkan-Loader.git" vulkan-loader; then
+        _shinchiro_patches="https://raw.githubusercontent.com/shinchiro/mpv-winbuild-cmake/master/packages"
+        do_uninstall "${_check[@]}"
+        do_patch "$_shinchiro_patches/vulkan-0001-cross-compile-static-linking-hacks.patch"
+        create_build_dir
+        log dependencies ../scripts/update_deps.py --no-build
```
Comment (wiiaboo) [inline build/media-suite_compile.sh]: Better explicitely run it with `/usr/bin/python3`.

Comment (wiiaboo) [reply build/media-suite_compile.sh]: mingw python3/python2 comes first in path and might not work.

Comment (1480c1) [inline build/media-suite_compile.sh]:
```suggestion
        log dependencies /mingw64/bin/python3 ../scripts/update_deps.py --no-build
```

Comment (1480c1) [reply build/media-suite_compile.sh]: wait, it is better to use msys2 python or mingw?

Comment (wiiaboo) [reply build/media-suite_compile.sh]: Msys2 one.

**Comment by wiiaboo** [comment]
Just use msys2 python and you don't have to worry about bitness either.

**Pull request merged**